project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,308 @@
+                            const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
+                            var messageID = decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex);
+                            //console.log(decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex));
+                            if (emailDateValue.indexOf("X-Notifications") != -1) {
+                                emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
+                            }
+                            //var emailObject = {};
+                            //emailObject[messageID] = {emailDate: emailDateValue, emailBody: emailBodyValue}; 
+                            if (emailBodyValue.indexOf("format=flowed") == -1 && emailDateValue.length > 1 && emailBodyValue.indexOf("Daily Insider") == -1) {
+                                emailObject[messageID] = {emailDate: emailDateValue, emailBody: emailBodyValue};
+                                list[list.length] = emailObject;
+                            }
+                            //console.log(emailBodyValue.indexOf())
+                            //console.log(emailBodyValue);
+                            //console.log(decodedEmail);
+                            //arrayList = list;
+                            totalObjects = emailObject;
+                            //console.log(decodedEmail);
+                            console.log(totalObjects);
+                            //console.log(list);
+                            const t1 = performance.now();
+                            console.log(t1-t0);
+                        });
+                    }
+                    }
+                });
+            };
+            //console.log(list);
+            //const t1 = performance.now();
+            //console.log(t1-t0);
+            var initialRequest = gapi.client.gmail.users.messages.list({
+            'userId': 'me',
+            'q': "Google Interview"
+        });
+        getPageOfMessages(initialRequest, []);
+        //console.log(list);
+        //const t1 = performance.now();
+        //console.log(t1-t0);
+    }
+
+    function getList() {
+        console.log(arrayList);
+    }
+
+    
+
+/**
+ * Get Message with given ID.
+ *
+ * @param  {String} userId User's email address. The special value 'me'
+ * can be used to indicate the authenticated user.
+ * @param  {String} messageId ID of Message to get.
+ * @param  {Function} callback Function to call when the request is complete.
+ */
+function getMessage(userId, messageId, callback) {
+  var request = gapi.client.gmail.users.messages.get({
+    'userId': userId,
+    'id': messageId
+  });
+  request.execute(callback);
+}
+
+/*************************************************************************************************************************** Brandon Vu Section*/
+
+
+
+
+
+
+var model;
```

Comment:
We can now use variables across different modules,  I got it to work with classes so removing this bottom part will be okay